test/com/google/javascript/jscomp/runtime_tests/utils/build.sh

```diff
@@ -19,31 +19,33 @@ else
   COMPILATION_LEVEL=$1
 fi
 
+# This directory.
+THIS_DIR=$(dirname $0)
+
+# Location of project root.
+PROJECT_ROOT=$(readlink -f $THIS_DIR/../../../../../../..)
+
 # Get the location of the local compiler in this directory, if it exists.
 # If it doesn't, build it, then resume execution.
-LOCAL_COMPILER="$(dirname ..)/target/closure-compiler-1.0-SNAPSHOT.jar"
+LOCAL_COMPILER="$PROJECT_ROOT/target/closure-compiler-1.0-SNAPSHOT.jar"
 if [ ! -f "$LOCAL_COMPILER" ]; then
   echo -e "\nCompiler JAR not built. Building...\n" && yarn build:fast
 fi
 
-# Translate a relative filepath to an absolute one.
-abs_dirname() {
-  echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
-}
-
-# Begin building runtime tests in the $TEST_DIR directory.
+# Build tests from the $TEST_DIR directory, where files like
+# `array_pattern_test.js` are stored.
 echo -e "\nBuilding runtime tests..."
-TEST_DIR="test/com/google/javascript/jscomp/runtime_tests"
+TEST_DIR="$THIS_DIR/.."
 
 # Get the absolute path of the test directory.
-ABS_PATH=$(abs_dirname "./$TEST_DIR")
+ABS_PATH=$(readlink -f $TEST_DIR)
```

Comment:
Nice!

Reply:
FYI, according to the man page for readlink

> Note realpath(1) is the preferred command to use for canonicalization functionality.

I think this is one of those situations where you can't win. :)
I've already imported this version, so no need to change this.